tests/checker/bad/gold/bad_map_key_1.scilla.gold

```diff
@@ -2,8 +2,7 @@
   "gas_remaining": "8000",
   "errors": [
     {
-      "error_message":
-        "This map type has an invalid key type. A closing bracket or proper separated capital identifier is expected.\n",
+      "error_message": "Syntax error, state number 86",
```

Comment:
It's a bit weird we have syntactic errors in the typechecker tests. Perhaps this could be refactored and moved into `tests/base/parser/bad` directory?

Reply:
Good point, but both this and the `list_to_map` case originate from `master`, so the problem should be fixed there rather than here on the `remote_state_reads` branch.

Reply:
Ah, I'm always missing the destination branch :)